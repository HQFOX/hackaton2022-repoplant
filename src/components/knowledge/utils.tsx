import {
  AssetInventoryMetadata,
  AssetInventoryConfiguration,
} from "@hitachivantara/uikit-react-core";

const metadata: AssetInventoryMetadata[] = [
  {
    id: "id1",
    accessor: "title",
    cellType: "alpha-numeric",
  },
  {
    id: "id2",
    accessor: "project",
    cellType: "alpha-numeric",
  },
];

const viewConfiguration: AssetInventoryConfiguration = {
  breakpoints: { xs: 12, sm: 6, md: 4, lg: 3, xl: 3 },
  columnConfiguration: [
    {
      style: { paddingLeft: "10px", textAlign: "start" },
    },
    {
      title: "Title",
      style: { textAlign: "start", minWidth: "120px" },
    },
    {
      title: "Project",
      style: { textAlign: "start", minWidth: "80px" },
    },
    {
      title: "Information Type",
      style: { textAlign: "start", minWidth: "60px" },
    },
    {
      title: "Uploaded By",
      style: { textAlign: "start", minWidth: "60px" },
    },
    {
      title: "Date",
      style: {
        textAlign: "start",
      },
    },
  ],
};

const configuration = {
  metadata,
  viewConfiguration,
};

const actions = [{ id: "view", label: "View", disabled: false }];

export { configuration, actions };
