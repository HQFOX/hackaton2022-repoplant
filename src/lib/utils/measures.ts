export const avg = (values: number[]): number => {
  if (!values || values.length === 0) {
    return 0;
  }
  return (
    values.reduce((a, b) => {
      return a + b;
    }) / values.length
  );
};
