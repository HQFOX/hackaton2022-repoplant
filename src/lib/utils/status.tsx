import React, { lazy } from "react";
import { Level5 } from "@hitachivantara/uikit-react-icons";
import { Semantic } from "@hitachivantara/uikit-react-lab";

export const getStatus = (status: any) => {
  const Level = `Level${status}`;
  const statusObj: {
    Icon: React.ReactNode;
    sema: Semantic;
  } = { Icon: <Level5 />, sema: "sema5" };
  return statusObj;
};
