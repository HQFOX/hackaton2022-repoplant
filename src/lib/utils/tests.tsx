import React from "react";
import { MemoryRouter } from "react-router-dom";
import GlobalProvider from "lib/providers/GlobalProvider";

export const withProvider = <P extends {}>(
  Component: React.ComponentType<P>
): React.FC<P> => (props) => (
  <MemoryRouter initialEntries={[{ pathname: "/", key: "testKey" }]}>
    <GlobalProvider>
      <Component {...props} />
    </GlobalProvider>
  </MemoryRouter>
);
