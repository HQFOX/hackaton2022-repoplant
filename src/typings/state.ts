import { AuthState } from "./auth";

export type DeepPartial<T> = T extends Function
  ? T
  : T extends object
  ? { [P in keyof T]?: DeepPartial<T[P]> }
  : T;

// The top-level state object
export interface AppState {
  readonly auth: AuthState;
}

export type TestAppState = DeepPartial<AppState>;
