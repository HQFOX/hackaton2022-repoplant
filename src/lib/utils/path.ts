import { Page } from "typings/pages";

const getSelection = (data: Page[], pathname: string): Page | undefined => {
  let selection;

  data.forEach((item) => {
    if (item.path === pathname) selection = { ...item };
    else if (item.data && pathname.indexOf(item.path) > -1)
      selection = getSelection(item.data, pathname);
  });

  return selection;
};

export { getSelection };
