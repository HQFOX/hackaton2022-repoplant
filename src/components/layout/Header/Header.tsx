import React, { useContext, useEffect, MouseEvent } from "react";
import { useTranslation } from "react-i18next";
import { useMediaQuery, useTheme } from "@material-ui/core";
import HvHeader, {
  HvHeaderBrand,
  HvHeaderActions,
  HvHeaderNavigation,
  NavigationItemProp,
} from "@hv/uikit-react-core/dist/Header";
import { HvButton } from "@hv/uikit-react-core";
import { LogOut, Menu, ThemeSwitcher, User } from "@hv/uikit-react-icons";
import HitachiLogo from "assets/HitachiLogo";
import { getSelection } from "lib/utils/path";
import ThemeContext from "lib/ThemeContext";
import NavigationContext from "lib/NavigationContext";
import { HeaderProps } from "./index";

const Header: React.FC<HeaderProps> = ({
  router,
  auth,
  pages,
  getPages,
  redirect,
  logout,
}: HeaderProps) => {
  const { t } = useTranslation();
  const theme = useTheme();
  const { isAuthed } = auth;
  const { pathname } = router.location;

  const { toggleTheme } = useContext(ThemeContext);
  const { toggleOpen } = useContext(NavigationContext);

  const isMdUp = useMediaQuery(theme.breakpoints.up("md"));
  const isXs = useMediaQuery(theme.breakpoints.only("xs"));

  const selection = getSelection(pages.data, pathname);

  useEffect(() => {
    getPages();
  }, [getPages]);

  const handleChange = (event: MouseEvent, s: NavigationItemProp): void => {
    if (s.path) redirect(s.path);
  };

  return pages?.data ? (
    <HvHeader>
      {!isMdUp && (
        <HvButton icon onClick={toggleOpen}>
          <Menu />
        </HvButton>
      )}

      <HvHeaderBrand
        logo={<HitachiLogo />}
        name={!isXs ? t("components.layout.header.appName") : undefined}
      />

      {isAuthed && isMdUp && (
        <HvHeaderNavigation
          data={pages.data}
          selected={selection?.id}
          onClick={handleChange}
        />
      )}

      <HvHeaderActions>
        <HvButton icon aria-label="Change theme" onClick={() => toggleTheme()}>
          <ThemeSwitcher />
        </HvButton>
        {isAuthed && (
          <HvButton icon aria-label="Open User panel">
            <User />
          </HvButton>
        )}
        {isAuthed && isMdUp && (
          <HvButton icon onClick={() => logout()} aria-label="Logout">
            <LogOut />
          </HvButton>
        )}
      </HvHeaderActions>
    </HvHeader>
  ) : (
    <></>
  );
};

export default Header;
