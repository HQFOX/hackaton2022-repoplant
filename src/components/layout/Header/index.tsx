import withStyles, { WithStyles } from "@material-ui/core/styles/withStyles";
import { Dispatch, bindActionCreators } from "redux";
import { connect } from "react-redux";
import { AppState } from "typings/state";
import { getPages, redirect } from "store/pages/thunks";
import { logout } from "store/auth/thunks";
import styles from "./styles";
import Header from "./Header";

const mapStateToProps = ({ router, auth, pages }: AppState) => ({
  router,
  auth,
  pages,
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      getPages,
      redirect,
      logout,
    },
    dispatch
  );

export type HeaderProps = WithStyles<typeof styles> &
  ReturnType<typeof mapStateToProps> &
  ReturnType<typeof mapDispatchToProps>;

export default withStyles(styles, { name: "Header" })(
  connect(mapStateToProps, mapDispatchToProps)(Header)
);
