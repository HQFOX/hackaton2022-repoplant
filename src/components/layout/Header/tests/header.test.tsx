import React from "react";
import { render } from '@testing-library/react';
import { withStoreProvider } from "lib/utils/tests";
import { TestAppState } from "typings/state";
import Header from "..";

describe("<Header />", () => {
  let component: any;

  beforeAll(() => {
    window.history.replaceState({ query: "google" }, "MOCK");
  });

  const WithStoreProvider = withStoreProvider(Header, {
    router: { location: { pathname: "/" } },
    auth: { isAuthed: false },
    pages: { data: [] },
    logout: jest.fn()
  } as TestAppState);

  beforeEach(() => {
    component = render(<WithStoreProvider />);
  });

  it("it matches the snapshot", () => {
    expect(component.container).toMatchSnapshot();
  });
});
