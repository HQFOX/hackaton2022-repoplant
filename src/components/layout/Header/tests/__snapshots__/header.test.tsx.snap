// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> it matches the snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root HvHeader-root MuiAppBar-positionFixed MuiAppBar-colorPrimary HvHeader-backgroundColor mui-fixed MuiPaper-elevation4"
  >
    <div
      class="HvHeader-header"
    >
      <button
        class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-text HvButton-icon HvButton-ghostSVG MuiButton-textPrimary HvButton-ghost"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <div
            class="HvIconBase-root HvIconBase-s"
            name="Menu"
          >
            <svg
              focusable="false"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="color0"
                d="M0 7.4h16v1H0zM0 3h16V2H0zm0 11h16v-1H0z"
                fill="#414141"
              />
            </svg>
          </div>
        </span>
      </button>
      <div
        class="HvHeaderBrand-root"
      >
        <svg
          height="16"
          style="width: 72px; height: 20px;"
          version="1.1"
          viewBox="0 0 80 16"
          width="80"
          x="0px"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
          y="0px"
        >
          <path
            d="M63.6,2.5c0,0,0,4.8,0,4.7H70c0,0,0-4.7,0-4.7s3.1,0,3.1,0c0,0,0,11.8,0,11.9H70c0,0,0-5.2,0-5.2c0,0-6.3,0-6.3,0c0,0,0,5.2,0,5.2h-3.1c0,0,0-11.9,0-11.9C60.5,2.5,63.6,2.5,63.6,2.5z"
            fill="#414141"
          />
          <path
            d="M33.9,2.5c0,0,0,2,0,2h-4.8v9.8h-3.1V4.5h-4.8v-2C21.1,2.5,33.9,2.5,33.9,2.5z"
            fill="#414141"
          />
          <path
            d="M46,14.3c0,0-3.5,0-3.5,0l-1-2.6h-5.9c0,0-1,2.6-1,2.6h-3.5l5.5-11.9h3.8L46,14.3z M38.6,4.5l-2.2,5.4h4.4L38.6,4.5"
            fill="#414141"
          />
          <rect
            fill="#414141"
            height="11.9"
            width="3.1"
            x="75.6"
            y="2.5"
          />
          <path
            d="M4.9,2.5c0,0,0,4.8,0,4.7h6.3c0,0,0-4.7,0-4.7s3.1,0,3.1,0c0,0,0,11.8,0,11.9h-3.1c0,0,0-5.2,0-5.2c0,0-6.3,0-6.3,0c0,0,0,5.2,0,5.2H1.7c0,0,0-11.9,0-11.9C1.7,2.5,4.9,2.5,4.9,2.5z"
            fill="#414141"
          />
          <rect
            fill="#414141"
            height="11.9"
            width="3.1"
            x="16.9"
            y="2.5"
          />
          <path
            d="M45.9,11c-0.3-0.8-0.4-1.6-0.4-2.5c0-1.2,0.2-2.4,0.8-3.4c0.6-1,1.5-1.8,2.7-2.2 c1.1-0.4,2.2-0.6,3.5-0.6c1.4,0,2.7,0.3,4,0.8c1.1,0.5,2,1.5,2.2,2.7c0.1,0.3,0.1,0.5,0.1,0.8h-3.3c0-0.3-0.1-0.6-0.2-0.9 c-0.3-0.6-0.8-1.2-1.5-1.4c-0.4-0.1-0.9-0.2-1.4-0.2c-0.5,0-1.1,0.1-1.5,0.3c-0.8,0.3-1.4,0.9-1.7,1.7 c-0.3,0.8-0.4,1.6-0.4,2.5c0,0.7,0.1,1.4,0.3,2.1c0.2,0.9,0.9,1.6,1.7,1.9c0.5,0.2,1.1,0.3,1.7,0.3c0.5,0,1-0.1,1.5-0.2 c0.6-0.2,1.1-0.6,1.4-1.2c0.2-0.4,0.3-0.8,0.3-1.3h3.3c0,0.4-0.1,0.8-0.2,1.1c-0.3,1.2-1.1,2.2-2.2,2.6 c-1.2,0.5-2.6,0.8-4,0.8c-1.1,0-2.2-0.2-3.2-0.5C47.8,13.6,46.5,12.5,45.9,11z"
            fill="#414141"
          />
        </svg>
        <div
          class="HvHeaderBrand-separator"
        />
        <p
          class="HvTypography-root HvTypography-highlightText"
        >
          components.layout.header.appName
        </p>
      </div>
      <div
        class="HvHeaderActions-root"
      >
        <button
          aria-label="Change theme"
          class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-text HvButton-icon HvButton-ghostSVG MuiButton-textPrimary HvButton-ghost"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <div
              class="HvIconBase-root HvIconBase-s"
              name="ThemeSwitcher"
            >
              <svg
                focusable="false"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="color0"
                  d="M14.54626 12.57422c.0268-.03833.05713-.07373.08326-.11231.11-.16284.20837-.33239.30584-.50244.03162-.05517.06751-.1073.09784-.16321.086-.15881.16034-.32287.23535-.48693.03364-.07349.07209-.144.10352-.21875.06787-.16126.1239-.32691.18109-.49244.02826-.08166.06165-.16088.08728-.24365.05506-.17773.09754-.359.14014-.54053.01745-.07409.04022-.14611.05554-.22119.0423-.20654.07123-.41626.09717-.62646.00659-.05371.01819-.10571.02374-.15967.02374-.23108.03369-.46435.03717-.69824.00061-.0365.00568-.07178.0058-.1084l-.00024-.00525.00067-.01428A7.98991 7.98991 0 008.001 0H8a8 8 0 000 16h.001a7.94253 7.94253 0 005.649-2.3501c.13953-.13891.26728-.28711.39533-.43506.03924-.04541.08276-.08691.121-.13317.13524-.16346.25872-.33467.37993-.50745zM8 15a6.953 6.953 0 01-4.943-2.057A7.02151 7.02151 0 018.001.95618V15z"
                  fill="#414141"
                />
              </svg>
            </div>
          </span>
        </button>
      </div>
    </div>
  </header>
</div>
`;
